README.md

```diff
@@ -1,57 +1,106 @@
 
-  http://www.apache.org/licenses/LICENSE-2.0
+    http://www.apache.org/licenses/LICENSE-2.0
 
 Unless required by applicable law or agreed to in writing, software
 distributed under the License is distributed on an "AS IS" BASIS,
 WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 
+=======
 
 # Gaffer Docker
 
 What you have here, is a Docker container for Gaffer.  This is a small
 instance and is really only useful only development / trial
 purposes.  But containerising allows a quick way to find out what Gaffer is
-like and develop against the interfaces.  The container deploys: Hadoop,
-Accumulo, Zookeeper, and launches Gaffer using Wildfly.
+like and develop against the interfaces.  To run Gaffer, you need four
+components:
+- Wildfly, hosting the Gaffer REST API and UI.
+- Accumulo, hosting the Gaffer extensions.
+- Zookeeper, which is used by Accumulo.
+- Hadoop, which is used by Accumulo.
+
+I have these components available as the four containers:
+- gchq/wildfly-gaffer
+- gchq/accumulo-gaffer
+- cybermaggedon/zookeeper
+- cybermaggedon/hadoop
+
+The code here:
+- Compiles Gaffer from source in a build container, and extracts a set of
+  things to be loaded into the deployable containers.
+- Downloads Wildfly, and creates a container integrating Wildfly and Gaffer.
+- Creates a container integrating Accumulo (cybermaggedon/accumulo) and
+  Gaffer.
+
+For Hadoop and Zookeeper, I have containers ready to use.
 
 The memory settings of Accumulo are low to ensure Accumulo runs in a
-small footprint.  Budget for around 2GB for trivial amounts of data, 6GB
-for anything with some load.
+small footprint.  Don't expect performance.
 
 To run:
 
-  docker run -it -p 8080:8080 gchq/gaffer:0.4.4
-
-You can then access the Gaffer API at port 8080:
-```
-http://localhost:8080/rest
 ```
+  # Run Hadoop
+  docker run --rm --name hadoop cybermaggedon/hadoop:2.7.3
+
+  # Run Zookeeper
+  docker run --rm --name zookeeper cybermaggedon/zookeeper:3.4.9
+
+  # Run Accumulo
+  docker run --rm --name accumulo --link zookeeper:zookeeper \
+        --link hadoop:hadoop gchq/accumulo-gaffer:0.4.4c
+
+  # Run Wildfly, exposing port 8080.
+  docker run --rm --name wildfly --link zookeeper:zookeeper \
+    --link hadoop:hadoop --link accumulo:accumulo \
+    -p 8080:8080 gchq/wildfly-gaffer:0.4.4c
 
-and the prototype Gaffer UI at:
-```
-http://localhost:8080/ui
 ```
 
+You can then access the Gaffer API at port 8080, e.g. try accessing URL
+http://HOSTNAME:8080/rest.  The UI is available at http://HOSTNAME:8080/ui.
 
 When the container dies, the data is lost.  If you want data to persist,
-mount a volume on /data e.g.
+put volumes on /data for Hadoop and Zookeeper, and /accumulo for Accumulo.
+Wildfly needs no persistent state.
 
-  docker run -it -p 8080:8080 -v /data/gaffer:/data gchq/gaffer:0.4.4
+```
```

Comment:
Would it be possible (or useful) to have a shell script that can launch everything in one go?